sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -98,19 +99,23 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("SPARK-18091: split large if expressions into blocks due to JVM code size limit") {
-    var strExpr: Expression = Literal("abc")
-    for (_ <- 1 to 150) {
-      strExpr = Decode(Encode(strExpr, "utf-8"), "utf-8")
-    }
+    // Set the max value at `WHOLESTAGE_HUGE_METHOD_LIMIT` to compile gen'd code by janino
+    withSQLConf(SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT.key -> Int.MaxValue.toString) {
```

Comment:
Why do we need to change value for `WHOLESTAGE_HUGE_METHOD_LIMIT` while this test is not for whole-stage codegen?
We could select better naming for `SQLConf.WHOLESTAGE_HUGE_METHOD_LIMIT`?

Reply:
yea, you're right. I think `CODEGEN_HUGE_METHOD_LIMIT` is better?